sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_additional_includes.py

```diff
@@ -133,6 +144,14 @@ def resolve(self) -> None:
         base_path = self._additional_includes_file_path.parent
         for additional_include in self._includes:
             src_path = (base_path / additional_include).resolve()
+            if self._zip_need_to_compress(src_path):
+                zip_additional_include = (base_path / additional_include).resolve()
+                folder_to_zip = zip_additional_include.parent / zip_additional_include.stem
+                with tempfile.TemporaryDirectory() as tmp_dir:
+                    tmp_folder_to_zip = (Path(tmp_dir) / "zip").resolve()
+                    shutil.copytree(folder_to_zip, tmp_folder_to_zip, ignore=shutil.ignore_patterns("__pycache__"))
+                    src_path = (Path(tempfile.mkdtemp()) / zip_additional_include.name).resolve()
+                    shutil.make_archive(str(src_path.parent / src_path.stem), "zip", tmp_folder_to_zip)
```

Comment:
So it will create a new zip file under src_path.parent? Is it expected? Or we just want to directly make archive under temp local code folder?

Reply:
it is expected; if directly make archive under temp folder, the upload code asset request's URL will change and break lots of tests, so I archive in one more deeper level.

Reply:
this is expected, the target of copy to temp folder is to filter all `__pycache__` inside the snapshot
I misunderstood this comment before, and I add some comments on this behavior.